src/service/url-replacements-impl.js

```diff
@@ -74,6 +74,22 @@ export class GlobalVariableSource extends VariableSource {
     /** @const {!./ampdoc-impl.AmpDoc} */
     this.ampdoc = ampdoc;
 
+    // A meta[name="amp-action-whitelist"] tag, if present, contains,
+    // in its content attribute, a whitelist of actions on the special AMP target.
+    if (this.ampVariableSubstitutionWhitelist_ === undefined
+      && this.ampdoc.getRootNode() && this.ampdoc.getRootNode().head) {
```

Comment:
The return value of `getRootNode()` is non-nullable and never `undefined`, so we shouldn't need to check `this.ampdoc.getRootNode()` ...